### ISSUE TITLE
A Rails Engine that overrides models from another engine isn't passing zeitwerk checks

### ISSUE THREAD

**Issue by dalton** [rails/rails#38935]
### Steps to reproduce

I am creating two engines. [Engine Two](https://github.com/ActiveTarget/ActiveAttack) builds on [Engine One](https://github.com/ActiveTarget/ActiveAttack). I am using both of these Engines in [an application](https://github.com/ActiveTarget/ActiveIntel) I'm building. I used the `class_eval` Engine [overriding technique](https://edgeguides.rubyonrails.org/engines.html#overriding-models-and-controllers) mentioned in the rails guide.

```ruby
# app/models/active_stix/tool_override.rb
ActiveStix::Tool.class_eval do
# Model override code here
end
```

The override is required from lib/engine_name/engine.rb as follows

```ruby
    initializer "active_attack.update_stix" do |app|
      engine_root = File.expand_path("../../..", __FILE__)
      Dir.glob(engine_root + "/app/overrides/**/*_override*.rb").each do |c|
        require_dependency(c)
      end
    end
```

### Expected behavior

Everything works as expected in development (overridden objects have the correct methods, attributes, and associations)

### Actual behavior
When I run `rails zeitwerk:check` or try to deploy the application to Heroku, I get the following error 

> expected file ~/.rvm/gems/ruby-2.7.0/gems/active_attack-0.1.14/app/overrides/models/active_stix/tool_override.rb to define constant Models::ActiveStix::ToolOverride

### System configuration

**Rails version**:

6.0.2

**Ruby version**:

2.7.0

This issue was previously posted as [a question on stack overflow](https://stackoverflow.com/questions/61157899/a-rails-engine-that-overrides-models-from-another-engine-isnt-passing-zeitwerk) but its similarity to  to [Rails issue 36100](https://github.com/rails/rails/issues/36100), led me to posting it here as well. That issue seems to describe how the main Rails app can override Engine models. It doesn't seem to work for one Engine overriding another Engine's models.

**Comment by fxn** [comment]
Zeitwerk assumes the files that are descendants to root directories define their expected constants. That is an auxiliary file, not meant to define a constant in your engine.

To support this use case, Zeitwerk allows you to opt-out by telling it to ignore the file:

```ruby
Rails.autoloaders.main.ignore(absolute/path/to/tool_override.rb)
```

Then, loading its contents is something you need to do by yourself because precisely Zeitwerk won't do it. And you do it right in the initializer.

In this case, however, there is no point in using `require_dependency` because this file is loaded exactly once on boot and no reloads are possible. Just issue a regular `require` call.

**Comment by fxn** [comment]
Ah, if everything in `app/overrides` plays the same role, you can ignore the whole directory too:

```ruby
Rails.autoloaders.main.ignore(absolute/path/to/app/overrides)
```

The API supports glob patterns as well.

**Comment by dalton** [comment]
Thanks! Doing this (in the Engine, right?) caused the following error

> NoMethodError: undefined method `eager_load_paths' for #<I18n::Config:0x00007fce6bb12c68>

But looking into that I discovered using `config.autoloader = :classic` met my immediate needs. I'll keep looking into zeitwerk to gain more familiarity, but my issue is currently resolved.

**Comment by fxn** [comment]
That is unrelated (see https://github.com/rails/rails/issues/37966), fix will ship with 6.0.3.

Thanks!
